build-system/babel-plugins/babel-plugin-transform-jss/index.js

```diff
@@ -86,12 +164,43 @@ module.exports = function ({template, types: t}) {
       },
 
       CallExpression(path, state) {
+        const callee = path.get('callee');
+
+        // Replace users that import JSS.
+        const importDeclaration = findImportDeclaration(callee, 'useStyles');
+        if (importDeclaration) {
+          const {parentPath} = path;
+          // Such as: useStyles().myClassName
+          if (replaceMemberExpression(importDeclaration, path, parentPath)) {
+            return;
+          }
+          // Such as: const classes = useStyles();
+          if (parentPath.isVariableDeclarator()) {
+            const {referencePaths} = path.scope.bindings[
+              parentPath.node.id.name
+            ];
+            const replacedReferenceCount = referencePaths.reduce(
+              (count, {parentPath}) =>
+                replaceMemberExpression(importDeclaration, path, parentPath)
+                  ? count + 1
+                  : count,
+              0
+            );
+            if (referencePaths.length === replacedReferenceCount) {
+              parentPath.remove();
+            }
+            return;
+          }
+          // Or unknown use, deopt.
```

Comment:
Wanna add destructure support? A long time ago I made sure destructures aren't deeply nested, so you'd only have to handle `const { foo } = useStyles()`